File: SSServer/src/main/java/com/prose/crhen/SSServer/dto/VolumeQueryDTO.java
package com.prose.crhen.SSServer.dto;

import com.prose.crhen.SSServer.model.Server;
import com.prose.crhen.SSServer.model.VolumeHistory;
import lombok.Builder;
import lombok.Value;

import java.sql.Date;
import java.util.Set;

@Value
@Builder
public class VolumeQueryDTO {
    
    String name;
    String desc;
    Date date;
    double fullCapacity;
    double latestStorageReserved;
    double latestStorageFree;
    double latestStorageRatio;
    String mappingLabel;
    Server server;
    Set<VolumeHistory> volumeHistories;

    public static class VolumeQueryDTOBuilder {
    }
    
}

File: SSKotlinServer/src/main/kotlin/com/prose/crhen/SSKotlinServer/SSKotlinServer/dto/VolumeQueryDTO.kt
package com.prose.crhen.SSKotlinServer.SSKotlinServer.dto

import com.prose.crhen.SSKotlinServer.SSKotlinServer.model.Server
import com.prose.crhen.SSKotlinServer.SSKotlinServer.model.VolumeHistory
import java.util.*

data class VolumeQueryDTO(
        val name: String,
        val desc: String,
        val date: Date,
        val fullCapacity: Double,
        val latestStorageReserved: Double,
        val latestStorageFree: Double,
        val latestStorageRatio: Double,
        val mappingLabel: String,
        val server: Server,
        val volumeHistories: Set<VolumeHistory>
)
